theano/tensor/nnet/abstract_conv.py

```diff
@@ -1520,7 +1517,7 @@ def flops(self, inp, outp):
             raise NotImplementedError(
                 'flops not implemented for convdim={}', self.convdim)
 
-    def conv(self, img, kern, mode="valid", dilation=1, unshared=False):
+    def conv(self, img, kern, mode="valid", dilation=1, unshared=False, direction=0):
```

Comment:
Could you change the `direction` to a string? (E.g., "forward", "backprop weights", "backprop inputs" as in BaseCorrMM.)